File: app/src/main/java/es/etologic/mahjongscoring2/app/game/activity/GameActivityModule.java
package es.etologic.mahjongscoring2.app.game.activity;

import dagger.Module;
import dagger.Provides;
import es.etologic.mahjongscoring2.domain.use_cases.CreateGameUseCase;
import es.etologic.mahjongscoring2.domain.use_cases.GetGamesUseCase;
import es.etologic.mahjongscoring2.domain.use_cases.UpdateGameUseCase;
import es.etologic.mahjongscoring2.domain.use_cases.UpdateRoundsUseCase;

@Module
public class GameActivityModule {

    @Provides
    GameActivityViewModelFactory provideGameViewModelFactory(CreateGameUseCase createGameUseCase,
                                                             GetGamesUseCase getGamesUseCase,
                                                             UpdateRoundsUseCase updateRoundUseCase,
                                                             UpdateGameUseCase updateGameUseCase) {
        return new GameActivityViewModelFactory(
                createGameUseCase,
                getGamesUseCase,
                updateRoundUseCase,
                updateGameUseCase);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/app/game/activity/GameActivityModule.kt
package es.etologic.mahjongscoring2.app.game.activity

import dagger.Module
import dagger.Provides
import es.etologic.mahjongscoring2.domain.use_cases.EndGameUseCase
import es.etologic.mahjongscoring2.domain.use_cases.GetCurrentGameUseCase
import es.etologic.mahjongscoring2.domain.use_cases.UpdateGameUseCase

@Module
class GameActivityModule {
    
    @Provides
    internal fun provideGameViewModelFactory(
        getCurrentGameUseCase: GetCurrentGameUseCase,
        updateGameUseCase: UpdateGameUseCase,
        endGameUseCase: EndGameUseCase
    ): GameActivityViewModelFactory {
        return GameActivityViewModelFactory(
            getCurrentGameUseCase,
            updateGameUseCase,
            endGameUseCase
        )
    }
}
